### PR TITLE
Feat/model import cmd

### LINKED ISSUE #1249: Model import command
Implement a command to help user import local gguf file to cortex.cpp
Epic #1154 

### CLI Setup

The `import` subcommand is added to the `models` command with the following options:

- `--model_id`: A required option to specify the unique identifier for the imported model
- `--model_path`: A required option to specify the absolute path to the GGUF model file

## Usage

To import a model, users can now use the following command:

```
./your_cli_app models import --model_id <unique_model_id> --model_path /path/to/your/model.gguf
```

### PR DESCRIPTION
**Pull request by nguyenhoangthuan99**
## Related Issues

Fix #1249 

# Add Model Import Command to CLI

This PR introduces a new `import` subcommand to the `models` command in our CLI application. The new functionality allows users to import GGUF (GPT-Generated Unified Format) models from local files into the application.

## Features

- Import GGUF models using a specified model ID and file path
- Automatically parse GGUF files to extract model configuration
- Generate and store YAML configuration files for imported models
- Add imported models to the model list for easy management

## Implementation Details

The implementation consists of two main parts:

1. `ModelImportCmd` class in `model_import_cmd.h` and `model_import_cmd.cpp`
2. CLI11 command setup in the main application file

### ModelImportCmd

This class handles the core logic of importing a model:

- Parses the GGUF file using `GGUFHandler`
- Creates a `ModelConfig` object with the parsed information
- Generates a YAML configuration file using `YamlHandler`
- Adds the model entry to the model list using `ModelListUtils`

Error handling is implemented to ensure robust operation, including cleaning up partially created files in case of failures.

### CLI Setup

The `import` subcommand is added to the `models` command with the following options:

- `--model_id`: A required option to specify the unique identifier for the imported model
- `--model_path`: A required option to specify the absolute path to the GGUF model file

## Usage

To import a model, users can now use the following command:

```
./cortex models import --model_id <unique_model_id> --model_path /path/to/your/model.gguf
```

### CODE REVIEW

**Comment by nguyenhoangthuan99** [comment]
Need to add API for model import, to allow Jan's user import their models

File: engine/controllers/command_line_parser.cc [hunk 1]
```diff
@@ -153,6 +154,20 @@ bool CommandLineParser::SetupCommand(int argc, char** argv) {
   auto model_update_cmd =
       models_cmd->add_subcommand("update", "Update configuration of a model");
 
+  std::string model_path;
+  auto model_import_cmd = models_cmd->add_subcommand(
+      "import", "Import a gguf model from local file");
+  model_import_cmd->add_option("--model_id", model_id, "");
+  model_import_cmd->require_option();
```
Comment (vansangpfiev) [inline engine/controllers/command_line_parser.cc]: Maybe should have a constraint with exact 2 required options

File: engine/commands/model_import_cmd.cc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                                 std::filesystem::path(model_handle_ + ".yml"))
+                                    .string();
+  modellist_utils::ModelEntry model_entry{
+      model_handle_,   "local",       "imported",
+      model_yaml_path, model_handle_, modellist_utils::ModelStatus::READY};
+  try {
+    std::filesystem::create_directories(
+        std::filesystem::path(model_yaml_path).parent_path());
+    gguf_handler.Parse(model_path_);
+    config::ModelConfig model_config = gguf_handler.GetModelConfig();
```
Comment (vansangpfiev) [inline engine/commands/model_import_cmd.cc]: nit: can use `auto`

File: engine/commands/model_import_cmd.cc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      yaml_handler.WriteYamlFile(model_yaml_path);
+      CLI_LOG("Model is imported successfully!");
+    } else {
+      CLI_LOG("Fail to import model, model_id '" + model_handle_ +
+              "' already exists!");
+    }
+
+  } catch (const std::exception& e) {
+    std::remove(model_yaml_path.c_str());
+    throw std::runtime_error("Error importing model '" + model_path_ +
```
Comment (vansangpfiev) [inline engine/commands/model_import_cmd.cc]: I don't see anything to catch this error yet

**Comment by dan-menlo** [comment]
> Need to add API for model import, to allow Jan's user import their models

@nguyenhoangthuan99 Will we be implementing the API as a separate issue?

**Review by dan-menlo — APPROVED** [review]
Lgtm - will let @vansangpfiev @namchuai evaluate for code. 

- Import API will be separate PR?

**Comment by nguyenhoangthuan99** [comment]
I added API in this PR because the logic is the same, and it makes us easier to track.

File: engine/e2e-test/test_cli_model_import.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+class TestCliModelImport:
+
+    @pytest.mark.skipif(True, reason="Expensive test. Only test when you have local gguf file.")
+    def test_model_import_should_be_success(self):
+
+        exit_code, output, error = run(
+            "Pull model", ["models", "import", "--model_id","test_model","--model_path","/path/to/local/gguf"],
+            timeout=None
+        )
+        assert exit_code == 0, f"Model import failed failed with error: {error}"
```
Comment (namchuai) [inline engine/e2e-test/test_cli_model_import.py:13]: hmm, if running in isolation, this test will failed, right?

File: engine/commands/model_import_cmd.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <cmath>  // For std::isnan
```
Comment (namchuai) [inline engine/commands/model_import_cmd.h]: we might not need this `#include`

File: engine/e2e-test/test_cli_model_import.py (hunk 4 above)
Comment (nguyenhoangthuan99) [reply engine/e2e-test/test_cli_model_import.py:13]: Yes, this test is used only if logics change for the model-import command.

File: engine/commands/model_import_cmd.h (hunk 5 above)
Comment (nguyenhoangthuan99) [reply engine/commands/model_import_cmd.h]: I'll update it

**Comment by dan-menlo** [comment]
Closes https://github.com/janhq/cortex.cpp/issues/1058

**Pull request merged**